src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -19,6 +19,50 @@
 #define UCT_DC_MLX5_IFACE_TXQP_GET(_iface, _ep, _txqp, _txwq) \
     UCT_DC_MLX5_IFACE_TXQP_DCI_GET(_iface, (_ep)->dci, _txqp, _txwq)
 
+
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_iface_inline_post(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep,
+                              uint8_t dci_index, unsigned opcode, unsigned length,
+                              /* SEND */ uint8_t am_id, uint64_t hdr, uint32_t imm_val_be,
+                              /* RDMA */ uint64_t rdma_raddr, uct_rkey_t rdma_rkey,
+                              uint8_t send_flags, const void *buffer)
+{
+    UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+
+    ucs_assert(!(iface->tx.dcis[dci_index].flags &
+                         UCT_DC_MLX5_DCI_FLAG_FAILED));
+
+    UCT_DC_MLX5_IFACE_TXQP_DCI_GET(iface, dci_index, txqp, txwq);
```

Comment:
maybe we move the assert flag to txwq or txqp and the assert check into txqp_inline_post, then no need to add these wrappers, and will check all mlx5 transports?

Reply:
done